src/caustics/utils.py

```diff
@@ -115,6 +116,97 @@ def get_meshgrid(
     )
     return torch.meshgrid([xs, ys], indexing="xy")
 
+@lru_cache(maxsize=32)
+def _quad_table(n, p, dtype, device):
+    """
+    from: https://pomax.github.io/bezierinfo/legendre-gauss.html
+
+    Args:
+        n (int): The number of quadrature points in each dimension.
+        p (Tensor): The pixelscale.
+        dtype (torch.dtype): The desired data type of the tensor.
+        device (torch.device): The device on which to create the tensor.
+
+    Returns:
+        Tuple[Tensor, Tensor, Tensor]: The generated meshgrid as a tuple of Tensors.
+    """
```

Comment:
Just noticed that this docstring style is not numpydoc. Please start following this guide https://numpydoc.readthedocs.io/en/latest/format.html so that we're all consistent 😄 Thanks!

Reply:
Good catch! Old habits die hard